### ISSUE TITLE
File type

### ISSUE THREAD

**Issue by Siddartha9889** [bluedenim/log4j-s3-search-samples#21]
Hi @bluedenim , as in the previous issue you have mentioned **"If you set the "s3Compression" property to "true" in the log4j2 config, then the contents will be GZIPed, and the S3 key will end in ".gz" to hint that the content is compressed".**
I have tried to set the compression to true, but the file is not ending with the .gz. Can you please provide more context on these , how we can set the file type. is there any release ?

And the another question how to reference the actual file that being uploaded to S3, it is just indicating with the time stamp-hostname-guid. how do we refer the actual file. Can you please provide more information on these.

**Comment by bluedenim** [comment]
Sorry, I gave you incorrect incomplete info before.

To compress the data and use the ".gz" suffix:
- enable the `s3Compression` option -- this will compress the contents before uploading
- enable the `s3KeyGzSuffixEnabled` option -- this will add the ".gz" suffix to the S3 key. NOTE that this is ignored if `s3Compression` is not enabled.

The entire list of options for S3 is documented here: https://github.com/bluedenim/log4j-s3-search?tab=readme-ov-file#s3

**Comment by bluedenim** [comment]
> how to reference the actual file that being uploaded to S3

Can you rephrase the question? I don't quite understand what you are asking.

**Comment by bluedenim** [comment]
Lastly, this repo is for the example program to test/try the [log4j-s3-search library](https://github.com/bluedenim/log4j-s3-search). If you have questions about the library, open an issue over there. 

I don't monitor this repo much. Thanks!

**Comment by bluedenim** [comment]
Closing due to inactivity.
